aait/backend/group-2B/Application/Features/Posts/Handlers/Commands/RemovePostCommandHandler.cs

```diff
@@ -0,0 +1,36 @@
+using AutoMapper;
+using MediatR;
+using SocialSync.Application.Contracts.Persistence;
+using SocialSync.Application.Features.Posts.Requests.Commands;
+using SocialSync.Application.Common.Responses;
+
+namespace SocialSync.Application.Features.Posts.Handlers.Commands;
+
+public class RemovePostCommandHandler : PostsRequestHandler, IRequestHandler<RemovePostCommand, BaseCommandResponse>
+{
+    public RemovePostCommandHandler(IUnitOfWork unitOfWork, IMapper mapper) : base(unitOfWork, mapper)
+    {
+    }
+
+    public async Task<BaseCommandResponse> Handle(RemovePostCommand request, CancellationToken cancellationToken)
+    {
+        var response = new BaseCommandResponse();
+
+
+        var post = await _postRepository.GetAsync(request.Id);
+        if (post != null)
+        {
+            request.Id = post.Id;
+            response.Success = true;
+            response.Message = "Post Deletion Successful";
+            await _postRepository.DeleteAsync(post);
```

Comment:
save the change to the database and check if the save is successful and only then you can make the result successfull